src/gui/owncloudgui.cpp

```diff
@@ -870,14 +870,26 @@ void ownCloudGui::slotShowShareDialog(const QString &sharePath, const QString &l
 
     const auto accountState = folder->accountState();
 
+    // As a first approximation, set the set of permissions that can be granted
+    // either to everything (resharing allowed) or nothing (no resharing).
+    //
+    // The correct value will be found with a propfind from ShareDialog.
+    SharePermissions maxSharingPermissions =
+            SharePermissions(SharePermissionRead)
```

Comment:
i believe you can use Q_DECLARE_OPERATORS_FOR_FLAGS  and then you don't need the cast